### ISSUE TITLE
net: "container:[name or id]" should support a named service and recognize the dependency

### ISSUE THREAD

**Issue by gilclark** [docker/compose#493]
Specifying net: "container:name" implies that the named container should be something fig recognizes as a service specified in the fig file.   It should also recognize that service as a dependency to be started first.

Example:
foo:
  image: kubernetes/pause:go
  ports:
    - "80:80"

bar:
  image: mynginx:latest
  net: "container: foo"

Right now this won't work because fig will look for the container literally named foo instead of the docker name actually given to it by fig.  Also fig doesn't know to start foo first because it doesn't know bar needs it.

Maybe something like;

bar:
  image: mynginx:latest
  net: "service:foo"

Thanks

**Comment by cboettig** [comment]
+1

We can currently get around this by using the name fig will give the container (e.g. `net: "container:dirname_foo_1"` and then adding a `volumes_from` to trick fig into establishing the dependency order correctly. 

It would be great if this issue could be addressed properly, but no python-foo here.

**Comment by gilclark** [comment]
Hi all,

I did implement this on my fork, https://github.com/gilclark/fig and it works great. The net container works and is recognized as a dependency so it starts ahead of it's dependents, like a link.  I also added the ability to specify dependencies explicitly even if containers weren't linked so that the order of startup could be guaranteed for any containers, linked or not.

The changes to fig are deep in a few places and so my pull requests have been ignored.  Kind of disappointing, but I've kept my fork up to date with the fig / docker-compose changes.

Maybe with enough votes the docker guys will take my pull request....

Thanks
